src/Submissions/Api/Frontend/js/pages/Admin/adac-blockallow-list.js

```diff
@@ -1,139 +1,157 @@
+
+  this.hideModal = function () {
+    $(_this.modalId).modal("hide");
+  };
+
+  this.openModalForAdd = function () {
+    _this.modal.mode(_this.modal.modalModeAdd);
+    _this.modal.registrationDomainRule(
+      new RegistrationDomainRule(0, "", "Block")
+    );
+
+    _this.showModal();
+  };
+
+  this.openModalForEdit = function (_, event) {
+    _this.modal.mode(_this.modal.modalModeEdit);
+
+    var registrationDomainRule = $(event.currentTarget).data("domain-rule");
+
+    _this.modal.registrationDomainRule(
+      new RegistrationDomainRule(
+        registrationDomainRule.id,
+        registrationDomainRule.value,
+        registrationDomainRule.ruleType
+      )
+    );
+
+    _this.showModal();
+
+    // turn off editing ruletype
+    $("#ruleTypes").find("input:radio:not(:checked)").attr("disabled", true);
+  };
+
+  this.modalSubmit = function (e) {
+    e.preventDefault();
+    var form = $(e.target); // get form as a jquery object
+
+    // Get Form Data
+    var data = serializeFormData(form);
+
+    // Get Action Type
+    var action = _this.modal.mode();
+    if (action == "Add") {
+      addRefData(
+        _this,
+        form.data("resource-url"),
+        data,
+        form.data("success-redirect"),
+        form.data("refdata-type")
+      ); // cf. adac-refdata-utility.js
+    } else if (action == "Update") {
+      editRefData(
+        _this,
+        form.data("resource-url") + "/" + $(e.target.Id).val(),
+        data,
+        form.data("success-redirect"),
+        form.data("refdata-type")
+      );
+    }
+  };
 }
 
 $(function () {
-    $("#modal-domain-rule-form").submit(function (e) {
-        adacRegistrationDomainRuleVM.modalSubmit(e);
-    });
-
-    $(".delete-confirm").click(function (e) {
-        e.preventDefault();
-
-        var $link = $(this);
-        var linkData = $link.data("domain-rule")
-        var url = $link.data("resource-url") + "/" + linkData.Id;
-
-        bootbox.confirm("Are you sure you want to delete " + linkData.Value + "?",
-            function (confirmation) {
-                if (confirmation) {
-                    deleteRefData(url, $link.data("success-redirect"), $link.data("refdata-type"));
-                }
-            }
-        );
-    });
-
-    adacRegistrationDomainRuleVM = new AdacRegistrationDomainRuleViewModel();
-    ko.applyBindings(adacRegistrationDomainRuleVM);
-
-    // re-enable radio buttons on modal close
-    $('#domain-rule-modal').on('hidden.bs.modal', function (e) {
-        $('#ruleTypes').find('input:radio').attr('disabled', false);
-    })
+  $("#modal-domain-rule-form").submit(function (e) {
+    adacRegistrationDomainRuleVM.modalSubmit(e);
+  });
+
+  $(".delete-confirm").click(function (e) {
+    e.preventDefault();
+
+    var $link = $(this);
+    var linkData = $link.data("domain-rule");
+    var url = $link.data("resource-url") + "/" + linkData.Id;
+
+    bootbox.confirm(
+      "Are you sure you want to delete " + linkData.Value + "?",
```

Comment:
Based on fixing the ref data patterns:

Does this work? (does the modal show the value of the rule?)
Fix is probably to lowercase if not:
```suggestion
      "Are you sure you want to delete " + linkData.value + "?",
```

And does it actually delete successfully? If not, again fix is lowercase on the linkData.id (line 104)